runtime/near-vm-runner/fuzz/fuzz_targets/runner.rs

```diff
@@ -0,0 +1,108 @@
+#![no_main]
+
+use arbitrary::Arbitrary;
+use core::fmt;
+use near_primitives::contract::ContractCode;
+use near_primitives::runtime::fees::RuntimeFeesConfig;
+use near_primitives::version::PROTOCOL_VERSION;
+use near_vm_logic::mocks::mock_external::MockedExternal;
+use near_vm_logic::{VMConfig, VMContext, VMOutcome};
+use near_vm_runner::internal::wasmparser::{Export, ExternalKind, Parser, Payload, TypeDef};
+use near_vm_runner::internal::VMKind;
+use near_vm_runner::VMError;
+
+libfuzzer_sys::fuzz_target!(|module: ArbitraryModule| {
+    let code = ContractCode::new(module.0.to_bytes(), None);
+    let (_outcome, _err) = run_fuzz(&code, VMKind::Wasmer0);
+});
+
+fn run_fuzz(code: &ContractCode, vm_kind: VMKind) -> (Option<VMOutcome>, Option<VMError>) {
+    let mut fake_external = MockedExternal::new();
+    let mut context = create_context(vec![]);
+    context.prepaid_gas = 10u64.pow(14);
+    let config = VMConfig::test();
+    let fees = RuntimeFeesConfig::test();
+
+    let promise_results = vec![];
+
+    let method_name = find_entry_point(code).unwrap_or_else(|| "main".to_string());
+    vm_kind.runtime().unwrap().run(
+        &code,
+        &method_name,
+        &mut fake_external,
+        context,
+        &config,
+        &fees,
+        &promise_results,
+        PROTOCOL_VERSION,
+        None,
+    )
+}
+
+/// Finds a no-parameter exported function, something like `(func (export "entry-point"))`.
+fn find_entry_point(contract: &ContractCode) -> Option<String> {
+    let mut tys = Vec::new();
+    let mut fns = Vec::new();
+    for payload in Parser::default().parse_all(contract.code()) {
+        match payload {
+            Ok(Payload::FunctionSection(rdr)) => fns.extend(rdr),
+            Ok(Payload::TypeSection(rdr)) => tys.extend(rdr),
+            Ok(Payload::ExportSection(rdr)) => {
+                for export in rdr {
+                    if let Ok(Export { field, kind: ExternalKind::Function, index }) = export {
+                        if let Some(&Ok(ty_index)) = fns.get(index as usize) {
+                            if let Some(Ok(TypeDef::Func(func_type))) = tys.get(ty_index as usize) {
+                                if func_type.params.is_empty() && func_type.returns.is_empty() {
+                                    return Some(field.to_string());
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            _ => (),
+        }
+    }
+    None
+}
+
+fn create_context(input: Vec<u8>) -> VMContext {
+    VMContext {
```

Comment:
Aren't some of these potentially controlled by the user? Would be ideal if the user-influenced parts were `Arbitrary` too. Potentially as a separate fuzz target?

Reply:
Yeah, that's a good idea. Let me issue this... #5465